From: "Arnd Bergmann" <arnd@arndb.de>
Subject: kexec-consolidate-kexec-and-crash-options-into-kernel-kconfigkexec-fix
Date: Thu, 06 Jul 2023 14:18:36 +0200

Today's linux-next now runs into a warning on arm64 and
presumably others, with the same problem as on arm earlier:

WARNING: unmet direct dependencies detected for KEXEC
  Depends on [n]: ARCH_SUPPORTS_KEXEC [=n]
  Selected by [y]:
  - CRASH_DUMP [=y] && ARCH_SUPPORTS_CRASH_DUMP [=y]

I think the easiest way to make this reliable would be
this fixup.

Link: https://lkml.kernel.org/r/c057ba7a-3cc2-44ac-a961-fa062b909876@app.fastmail.com
Signed-off-by: "Arnd Bergmann" <arnd@arndb.de>
Cc: Eric DeVolder <eric.devolder@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/Kconfig.kexec |    1 +
 1 file changed, 1 insertion(+)

--- a/kernel/Kconfig.kexec~kexec-consolidate-kexec-and-crash-options-into-kernel-kconfigkexec-fix
+++ a/kernel/Kconfig.kexec
@@ -91,6 +91,7 @@ config KEXEC_JUMP
 config CRASH_DUMP
 	bool "kernel crash dumps"
 	depends on ARCH_SUPPORTS_CRASH_DUMP
+	depends on ARCH_SUPPORTS_KEXEC
 	select CRASH_CORE
 	select KEXEC
 	help
_
